libs/langchain/langchain/smith/evaluation/string_run_evaluator.py

```diff
@@ -145,11 +145,9 @@ def map(self, run: Run) -> Dict[str, str]:
         """Maps the Run to a dictionary."""
         if not run.outputs:
             raise ValueError(f"Run {run.id} has no outputs to evaluate.")
-        if run.run_type != "chain":
-            raise ValueError("Chain RunMapper only supports Chain runs.")
-        if self.input_key not in run.inputs:
+        if self.input_key is not None and self.input_key not in run.inputs:
             raise ValueError(f"Run {run.id} does not have input key {self.input_key}.")
-        elif self.prediction_key not in run.outputs:
+        elif self.prediction_key is not None and self.prediction_key not in run.outputs:
```

Comment:
code seems to do auto behavior with sources being dict of size 1, would be helpful to have more of the intent documented since it's unclear whether the auto behavior is good or bad

Reply:
Ya auto behavior if it's a dict of size 1 and there's no key manually provided. I'll doc